### ISSUE TITLE
Order has shipping address fields when shipping is disabled in settings

### ISSUE THREAD

**Issue by strangerkir** [woocommerce/woocommerce#37478]
### Prerequisites

- [ ] I have carried out troubleshooting steps and I believe I have found a bug.
- [X] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

When shipping is disabled in WooCommerce settings, manually created order still has editable shipping fields. But in the same time, there are no available options for country select. This is confusing and may lead to inconsistent data if merchant is not careful enough.

### Expected behavior

There are no shipping fields on the order edit page when shipping is disabled in the WooCommerce settings. At least, there should be no such fields for newly created orders.

### Actual behavior

Shipping fields are editable for the new manually created order, although there is no available country and it doesn't make sense in general.

### Steps to reproduce

1. Go to WooCommerce-> Settings -> General.
2. Select `Disable shipping & shipping calculations` for `Shipping location(s)`.
3. Save settings.
4. Go to the WooCommerce-> Orders.
5. Click `Add Order`.
6. Click the pencil icon to edit shipping address.
7. Observe shipping address is editable.
8. Try to set any country.
9. Observe there are no available countries.

### WordPress Environment

\`

### WordPress Environment

WordPress address (URL): [http://woocommerce.ddev.site](<http://woocommerce.ddev.site>)<br>Site address (URL): [http://woocommerce.ddev.site](<http://woocommerce.ddev.site>)<br>WC Version: 7.5.1<br>REST API Version: ✔ 7.5.1<br>WC Blocks Version: ✔ 9.6.6<br>Action Scheduler Version: ✔ 3.5.4<br>Log Directory Writable: ✔<br>WP Version: 6.1.1<br>WP Multisite: –<br>WP Memory Limit: 1 GB<br>WP Debug Mode: ✔<br>WP Cron: ✔<br>Language: en_US<br>External object cache: –

### Server Environment

Server Info: nginx/1.22.1<br>PHP Version: 8.0.28<br>PHP Post Max Size: 100 MB<br>PHP Time Limit: 600<br>PHP Max Input Vars: 5000<br>cURL Version: 7.74.0<br>OpenSSL/1.1.1n

SUHOSIN Installed: –<br>MySQL Version: 10.4.28-MariaDB-1:10.4.28+maria\~ubu2004-log<br>Max Upload Size: 100 MB<br>Default Timezone is UTC: ✔<br>fsockopen/cURL: ✔<br>SoapClient: ✔<br>DOMDocument: ✔<br>GZip: ✔<br>Multibyte String: ✔<br>Remote Post: ✔<br>Remote Get: ✔

### Database

WC Database Version: 7.5.1<br>WC Database Prefix: wp\_<br>Total Database Size: 3.85MB<br>Database Data Size: 2.41MB<br>Database Index Size: 1.44MB<br>wp_woocommerce_sessions: Data: 0.02MB + Index: 0.02MB + Engine InnoDB<br>wp_woocommerce_api_keys: Data: 0.02MB + Index: 0.03MB + Engine InnoDB<br>wp_woocommerce_attribute_taxonomies: Data: 0.02MB + Index: 0.02MB + Engine InnoDB<br>wp_woocommerce_downloadable_product_permissions: Data: 0.02MB + Index: 0.06MB + Engine InnoDB<br>wp_woocommerce_order_items: Data: 0.02MB + Index: 0.02MB + Engine InnoDB<br>wp_woocommerce_order_itemmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB<br>wp_woocommerce_tax_rates: Data: 0.02MB + Index: 0.06MB + Engine InnoDB<br>wp_woocommerce_tax_rate_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB<br>wp_woocommerce_shipping_zones: Data: 0.02MB + Index: 0.00MB + Engine InnoDB<br>wp_woocommerce_shipping_zone_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB<br>wp_woocommerce_shipping_zone_methods: Data: 0.02MB + Index: 0.00MB + Engine InnoDB<br>wp_woocommerce_payment_tokens: Data: 0.02MB + Index: 0.02MB + Engine InnoDB<br>wp_woocommerce_payment_tokenmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB<br>wp_woocommerce_log: Data: 0.02MB + Index: 0.02MB + Engine InnoDB<br>wp_actionscheduler_actions: Data: 0.02MB + Index: 0.11MB + Engine InnoDB<br>wp_actionscheduler_claims: Data: 0.02MB + Index: 0.02MB + Engine InnoDB<br>wp_actionscheduler_groups: Data: 0.02MB + Index: 0.02MB + Engine InnoDB<br>wp_actionscheduler_logs: Data: 0.02MB + Index: 0.03MB + Engine InnoDB<br>wp_commentmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB<br>wp_comments: Data: 0.02MB + Index: 0.09MB + Engine InnoDB<br>wp_links: Data: 0.02MB + Index: 0.02MB + Engine InnoDB<br>wp_options: Data: 1.48MB + Index: 0.03MB + Engine InnoDB<br>wp_postmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB<br>wp_posts: Data: 0.02MB + Index: 0.06MB + Engine InnoDB<br>wp_termmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB<br>wp_terms: Data: 0.02MB + Index: 0.03MB + Engine InnoDB<br>wp_term_relationships: Data: 0.02MB + Index: 0.02MB + Engine InnoDB<br>wp_term_taxonomy: Data: 0.02MB + Index: 0.03MB + Engine InnoDB<br>wp_usermeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB<br>wp_users: Data: 0.02MB + Index: 0.05MB + Engine InnoDB<br>wp_wc_admin_notes: Data: 0.05MB + Index: 0.00MB + Engine InnoDB<br>wp_wc_admin_note_actions: Data: 0.02MB + Index: 0.02MB + Engine InnoDB<br>wp_wc_category_lookup: Data: 0.02MB + Index: 0.00MB + Engine InnoDB<br>wp_wc_customer_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB<br>wp_wc_download_log: Data: 0.02MB + Index: 0.03MB + Engine InnoDB<br>wp_wc_order_coupon_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB<br>wp_wc_order_product_lookup: Data: 0.02MB + Index: 0.06MB + Engine InnoDB<br>wp_wc_order_stats: Data: 0.02MB + Index: 0.05MB + Engine InnoDB<br>wp_wc_order_tax_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB<br>wp_wc_product_attributes_lookup: Data: 0.02MB + Index: 0.02MB + Engine InnoDB<br>wp_wc_product_download_directories: Data: 0.02MB + Index: 0.02MB + Engine InnoDB<br>wp_wc_product_meta_lookup: Data: 0.02MB + Index: 0.09MB + Engine InnoDB<br>wp_wc_rate_limits: Data: 0.02MB + Index: 0.02MB + Engine InnoDB<br>wp_wc_reserved_stock: Data: 0.02MB + Index: 0.00MB + Engine InnoDB<br>wp_wc_tax_rate_classes: Data: 0.02MB + Index: 0.02MB + Engine InnoDB<br>wp_wc_webhooks: Data: 0.02MB + Index: 0.02MB + Engine InnoDB

### Post Type Counts

attachment: 1<br>page: 7<br>post: 1<br>shop_order: 1

### Security

Secure connection (HTTPS): ❌<br>					Your store is not using HTTPS. Learn more about HTTPS and SSL Certificates.<br>Hide errors from visitors: ❌Error messages should not be shown to visitors.

### Active Plugins (1)

WooCommerce: by Automattic – 7.5.1

### Inactive Plugins (2)

Akismet Anti-Spam: by Automattic – 5.0.1<br>Hello Dolly: by Matt Mullenweg – 1.7.2

### Settings

API Enabled: –<br>Force SSL: –<br>Currency: USD ($)<br>Currency Position: left<br>Thousand Separator: ,<br>Decimal Separator: .<br>Number of Decimals: 2<br>Taxonomies: Product Types: external (external)<br>grouped (grouped)<br>simple (simple)<br>variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)<br>exclude-from-search (exclude-from-search)<br>featured (featured)<br>outofstock (outofstock)<br>rated-1 (rated-1)<br>rated-2 (rated-2)<br>rated-3 (rated-3)<br>rated-4 (rated-4)<br>rated-5 (rated-5)

Connected to [WooCommerce.com](<http://WooCommerce.com>): –<br>Enforce Approved Product Download Directories: ✔<br>Order datastore: WC_Order_Data_Store_CPT

### WC Pages

Shop base: #5 - /shop/<br>Cart: #6 - /cart/<br>Checkout: #7 - /checkout/<br>My account: #8 - /my-account/<br>Terms and conditions: ❌ Page not set

### Theme

Name: Twenty Twenty-Three<br>Version: 1.0<br>Author URL: [https://wordpress.org](<https://wordpress.org>)<br>Child Theme: ❌ – If you are modifying WooCommerce on a parent theme that you did not build personally we recommend using a child theme. See: How to create a child theme<br>WooCommerce Support: ✔

### Templates

Overrides: –

### Admin

Enabled Features: activity-panels<br>analytics<br>coupons<br>customer-effort-score-tracks<br>import-products-task<br>experimental-fashion-sample-products<br>shipping-smart-defaults<br>shipping-setting-tour<br>homescreen<br>marketing<br>multichannel-marketing<br>mobile-app-banner<br>navigation<br>onboarding<br>onboarding-tasks<br>remote-inbox-notifications<br>remote-free-extensions<br>payment-gateway-suggestions<br>shipping-label-banner<br>subscriptions<br>store-alerts<br>transient-notices<br>woo-mobile-welcome<br>wc-pay-promotion<br>wc-pay-welcome-page

Disabled Features: minified-js<br>new-product-management-experience<br>product-variation-management<br>settings

Daily Cron: ✔ Next scheduled: 2023-03-29 16:44:45 +00:00<br>Options: ✔<br>Notes: 36<br>Onboarding: skipped

### Action Scheduler

Complete: 1<br>Oldest: 2023-03-28 16:44:53 +0000<br>Newest: 2023-03-28 16:44:53 +0000

Pending: 3<br>Oldest: 2023-03-28 16:45:18 +0000<br>Newest: 2023-03-29 16:44:53 +0000

### Status report information

Generated at: 2023-03-28 16:45:46 +00:00

### Isolating the problem

- [X] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [X] This bug happens with a default WordPress theme active, or [Storefront](<https://woocommerce.com/storefront/>).
- [X] I can reproduce this bug consistently using the steps above.

**Comment by barryhughes** [comment]
Thanks, we see the same thing. It definitely seems something needs to change here, though a possible alternative is making it so that the country selector (in the order editor billing fields) works properly in this context, following the general philosophy of allowing merchants complete control in this area.

**Comment by flakas** [comment]
Thanks for the report!

it's likely that #56922 made Barry's workaround work by making the country selector work, but it's a good design question - should shipping fields appear at all when shipping is disabled?

Assigning for design team's review.

**Comment by JanaMW27** [comment]
Hello 👋 I have been looking at this one from the design side. 

Good news first: the country dropdown is fine now

The report mentions the country dropdown having no options when shipping is disabled. That got fixed along the way in [#56922](https://github.com/woocommerce/woocommerce/pull/56922) back in April 2025, and it now lists every country and selects normally. I double-checked on a store with shipping switched off.

<img width="1246" height="722" alt="Image" src="https://github.com/user-attachments/assets/1dbace8e-5df6-4a20-969b-ecaaa3c4deab" />

What is left

With shipping disabled, a new order still offers a full shipping address form. The store does not ship anything, but you can fill an address in and save it.

Proposed direction

Two cases:

- Order has no shipping address. When shipping is disabled, hide the address fields, along with Edit, Load shipping address and Copy billing address, since all three open the same form. In their place, a short line: "Shipping is disabled for this store, shipping details cannot be edited. Enable shipping to add shipping information.", with a link to the setting.

<img width="1160" height="417" alt="Image" src="https://github.com/user-attachments/assets/3b6dd817-501c-4521-abeb-522f0f81cff0" />

- Order already has a shipping address. Leave it exactly as it works today, fully editable, no notice. Shipping gets switched off temporarily sometimes, and if an order already carries a real address a merchant should always be able to correct it.

Switching the setting back on restores everything by itself.

An issue: the customer note

This is the part I would most like a second opinion on.

The customer note currently sits inside the shipping form. It is not shipping information, it is what the shopper typed at checkout, but in the admin it lives in the same block as the address fields. So if the form goes, editing the note goes with it.

At checkout the address fields disappear when shipping is not needed, but the note stays in its own section with its own heading. 

Three options:

1. Leave the note where it is and always show it as an open text box. Simplest, though the same field then behaves differently depending on the store.
2. Show the note as text with its own small edit pencil, the way Billing already works. Consistent everywhere.
3. Move the note out of the Shipping column into its own section, mirroring the front end. Probably the tidiest in the long run, but it changes the screen for every store rather than only the ones with shipping off.

My preference is 2, because it's consistent with checkout and the rest of the order page.

Let me know your thoughts.

**Comment by opr** [comment]
Hey @JanaMW27 - Could we just move the customer note outside the shipping form and treat it as order-level information everywhere?

When saving, it is already processed independently of the shipping fields, so this would keep the fix straightforward and avoid special-case behaviour when shipping is disabled.

I don't think this would result in a visual change for merchants with shipping on or off.

**Comment by JanaMW27** [comment]
I think that would be ok, but what would we show in the Shipment section?
This?

<img width="338" height="95" alt="Image" src="https://github.com/user-attachments/assets/db3838b8-1621-41bf-97f9-a68318f57114" />

Without the edit icon?

Also, what are your thoughts on how to solve the Customer note issue?

**Comment by opr** [comment]
I feel that completely preventing the merchant from editing shipping isn't right, I think the example above, but *with* the icon makes sense.

> Also, what are your thoughts on how to solve the Customer note issue?

sorry if my answer above was not clear, but I propose moving it out of the Shipping form, I don't think this would result in any noticeable change for the merchant, it'd just change at a code-level making it easier to hide the shipping fields without affecting customer note.

**Comment by JanaMW27** [comment]
> I feel that completely preventing the merchant from editing shipping isn't right, I think the example above, but with the icon makes sense.

In that case, I disagree. The message "Shipping is disabled" and the edit icon to add shipping next to it seem contradictory. I can see a merchant scratching their head and wondering whether shipping is truly disabled.

If a merchant needs shipping details on an order, the fix is to turn shipping on, and the notice links to that setting. One click and everything comes back, including the icon.

I may be missing a case. Is there a real scenario where a merchant needs to add a shipping address while the store does not ship?

pinging @jamesckemp and @j111q  for extra input on this

> sorry if my answer above was not clear, but I propose moving it out of the Shipping form, I don't think this would result in any noticeable change for the merchant, it'd just change at a code-level making it easier to hide the shipping fields without affecting customer note.

Sounds good

**Comment by opr** [comment]
> In that case, I disagree. The message "Shipping is disabled" and the edit icon to add shipping next to it seem contradictory. I can see a merchant scratching their head and wondering whether shipping is truly disabled.
>
> If a merchant needs shipping details on an order, the fix is to turn shipping on, and the notice links to that setting. One click and everything comes back, including the icon.

OK, let's hear from product, that would be helpful. In any case, I'm happy to go with whatever decision you land on 👍🏼

**Comment by j111q** [comment]
(Chiming in with my 2 cents, let's see what @jamesckemp says also.)

> I may be missing a case. Is there a real scenario where a merchant needs to add a shipping address while the store does not ship?

@JanaMW27 , I _have_ heard of B2B stores running stuff like price-less, quote-based catalogs, so they may have some products which can't be added to cart and shipping as a feature is not set up for the store, but a shipping address CAN be added manually into an order, and products will be shipped there... I don't know how common a situation it is but I have seen it at least twice during customer interviews. 

**TLDR I would actually hesitate to completely disable the Shipping fields even if Shipping is not enabled for a store.**

Perhaps we can leave the edit icon, and in the notice, it might say something like:

`Shipping has not been enabled for this store. Manually add a shipping address, or enable shipping in Settings`

**Comment by JanaMW27** [comment]
OOOOOooo thank you @j111q, that is a valuable use case.
Then I think yes, let's keep the icon to edit the Shipping fields 👍.

And yes to the new messaging:
`Shipping has not been enabled for this store. Manually add a shipping address, or enable shipping in Settings`

cc @opr since we have alignment
